### PR TITLE
dbus: Update HEAD build

### PR DESCRIPTION
**Pull request by ts826848**
./autogen took the place of ./configure. This updates that command and adds necessary depends_on.

### CODE REVIEW

File: Formula/dbus.rb [hunk 1]
```diff
@@ -30,16 +36,16 @@ def install
...
-                          "--prefix=#{prefix}",
-                          "--localstatedir=#{var}",
-                          "--sysconfdir=#{etc}",
-                          "--disable-xml-docs",
-                          "--disable-doxygen-docs",
-                          "--enable-launchd",
-                          "--with-launchd-agent-dir=#{prefix}",
-                          "--without-x",
-                          "--disable-tests"
+    system "./autogen.sh", "--disable-dependency-tracking",
```
Comment (MikeMcQuaid) [inline Formula/dbus.rb]: This change needs to be only for `build.head?`

Comment (aw1621107) [inline Formula/dbus.rb]: It appears so. If anything, it looks like the stable version (1.10.8 on my machine) made this change too, or at least it is an option for building.

Comment (ilovezfs) [inline Formula/dbus.rb]: @aw1621107 configure is generated by autogen. Sometimes a "release tarball" will already have it pre-generated for you. It's a distinction between building from the raw source code repository and building from a packaged release of the sources.
```
Josephs-MacBook-Pro:~ joe$ tar -tf <(curl -sL https://dbus.freedesktop.org/releases/dbus/dbus-1.10.8.tar.gz)|grep configure
dbus-1.10.8/configure.ac
dbus-1.10.8/configure
Josephs-MacBook-Pro:~ joe$ tar -tf <(curl -sL https://dbus.freedesktop.org/releases/dbus/dbus-1.11.4.tar.gz)|grep configure
dbus-1.11.4/configure.ac
dbus-1.11.4/configure
```
Whereas in the git repository:
```
$ ls dbus|grep configure
configure.ac
```
After running autogen, you will have configure just like the tarball.

Comment (aw1621107) [inline Formula/dbus.rb]: @MikeMcQuaid Just realized that I read your sentence with the question mark as part of the sentence and not the code bit. HEAD builds are the only ones that **require** `./autogen.sh`, but it seems to work for the stable and devel versions too.

@ilovezfs Yeah, that makes sense as to why only HEAD builds require it.

**Comment by ilovezfs** [comment]
Passing on this for now, sorry. Thanks for the PR nonetheless!

**Comment by aw1621107** [comment]
:(

**Comment by ilovezfs** [comment]
@aw1621107 I don't mind reopening. Just make sure to limit the change to head only, and pass the arguments to configure not autogen.sh

Here's an example: https://github.com/Homebrew/homebrew-core/blob/a9e17ebc4754bb4f9e707eb76f1b7477044a6a24/Formula/pcre.rb

**Comment by aw1621107** [comment]
@ilovezfs The reason I passed the arguments to autogen is because it told me to when I ran it as just `./autogen.sh`. Something like

```
==> ./autogen.sh
I am going to run ./configure with no arguments - if you wish
to pass any to it, please specify them on the ./autogen.sh command line.
```

And I didn't need to invoke `./configure` at all -- seems like autogen.sh does that automatically? -- so that's why ./configure is gone

**Comment by ilovezfs** [comment]
@aw1621107 https://cgit.freedesktop.org/dbus/dbus/tree/autogen.sh

```
if test x"$NOCONFIGURE" = x; then
  run_configure=true
  for arg in $*; do
    case $arg in 
        --no-configure)
            run_configure=false
            ;;
        *)
            ;;
    esac
  done
else
  run_configure=false
fi
```
So pass `--no-configure` to autogen.sh

**Comment by aw1621107** [comment]
Will do then

**Comment by aw1621107** [comment]
And done!

**Comment by ilovezfs** [comment]
@aw1621107 looks great! Thanks for your persistence here! :+1:

**Pull request merged**